### PR TITLE
Update golang text dependency

### PR DESCRIPTION
**Pull request by ruflin**

### CODE REVIEW

File: filebeat/harvester/reader/line.go [hunk 1]
```diff
@@ -62,6 +62,12 @@ func (l *Line) Next() ([]byte, int, error) {
 		// Check last decoded byte really being '\n' also unencoded
 		// if not, continue reading
 		buf := l.outBuffer.Bytes()
+
+		// This can happen if something goes wrong during decoding
+		if len(buf) == 0 {
+			logp.Debug("wwer", "empty buffer, continue")
```
Comment (urso) [inline filebeat/harvester/reader/line.go]: wwer?

**Comment by ruflin** [comment]
@urso I now cleaned up the PR and added comments. I also updated to the most recent version. Please have a look.

File: filebeat/harvester/encoding/mixed.go [hunk 2]
```diff
@@ -1,16 +1,17 @@
 package encoding
 
-import "golang.org/x/text/transform"
+import "golang.org/x/text/encoding"
 
-type mixedEncoding struct {
-	decoder func() transform.Transformer
-	encoder func() transform.Transformer
-}
+// mixed incoder is a copy of encoding.Replacement
```
Comment (urso) [inline filebeat/harvester/encoding/mixed.go]: s/mixed incoder/mixed encoder/

File: filebeat/harvester/reader/line.go [hunk 3]
```diff
@@ -146,6 +157,9 @@ func (l *Line) decode(end int) (int, error) {
 		var nDst, nSrc int
 
 		nDst, nSrc, err = l.decoder.Transform(buffer, inBytes[start:end], false)
+		logp.Debug("werew", "decoder in: %+v", string(inBytes[start:end]))
+		logp.Debug("werew", "decoder out: %+v", string(buffer))
```
Comment (urso) [inline filebeat/harvester/reader/line.go]: werew?

File: filebeat/harvester/reader/line.go [hunk 4]
```diff
@@ -121,6 +129,9 @@ func (l *Line) advance() error {
 	// found encoded byte sequence for '\n' in buffer
 	// -> decode input sequence into outBuffer
 	sz, err := l.decode(idx + len(l.nl))
+	if err != nil {
+		logp.Err("Error decoding line: %s", err)
+	}
```
Comment (urso) [inline filebeat/harvester/reader/line.go]: decode method should reset `err=nil` on `transform.ErrShortDst`. `ErrShortDst` is transformer complaining about output buffer being full+more bytes in input buffer available for transcoding.

File: filebeat/harvester/encoding/mixed.go (hunk 2 above)
Comment (ruflin) [inline filebeat/harvester/encoding/mixed.go]: fixed

File: filebeat/harvester/reader/line.go (hunk 3 above)
Comment (ruflin) [inline filebeat/harvester/reader/line.go]: Removed

File: filebeat/harvester/reader/line.go (hunk 4 above)
Comment (ruflin) [inline filebeat/harvester/reader/line.go]: Fixed

**Comment by ruflin** [comment]
@urso New version with review changes pushed.

**Comment by urso** [comment]
LGTM

**Pull request merged**